### PR TITLE
Convert restlet 1.1 tests from groovy to java

### PR DESCRIPTION
**Pull request by laurit**
part of https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7195

### CODE REVIEW

File: instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      return controller(
+          ERROR,
+          () -> {
+            getResponse().setStatus(Status.valueOf(ERROR.getStatus()), ERROR.getBody());
+            return new StringRepresentation(ERROR.getBody());
+          });
+    }
+  }
+
+  public static class ExceptionResource extends BaseResource {
```
Comment (heyams) [inline instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java:61]:
```suggestion
  static class ExceptionResource extends BaseResource {
```

File: instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    public Representation represent(Variant variant) {
+      return controller(
+          EXCEPTION,
+          () -> {
+            throw new IllegalStateException(EXCEPTION.getBody());
+          });
+    }
+  }
+
+  public static class QueryParamResource extends BaseResource {
```
Comment (heyams) [inline instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java:73]:
```suggestion
  static class QueryParamResource extends BaseResource {
```

File: instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    @Override
+    public Representation represent(Variant variant) {
+      return controller(QUERY_PARAM, () -> new StringRepresentation(QUERY_PARAM.getBody()));
+    }
+  }
+
+  public static class PathParamResource extends BaseResource {
```
Comment (heyams) [inline instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java:81]:
```suggestion
  static class PathParamResource extends BaseResource {
```

File: instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+    @Override
+    public Representation represent(Variant variant) {
+      return controller(PATH_PARAM, () -> new StringRepresentation(PATH_PARAM.getBody()));
+    }
+  }
+
+  public static class RedirectResource extends BaseResource {
```
Comment (heyams) [inline instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java:89]:
```suggestion
  static class RedirectResource extends BaseResource {
```

File: instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+          REDIRECT,
+          () -> {
+            getResponse().setLocationRef(new Reference(REDIRECT.getBody()));
+            getResponse().setStatus(Status.valueOf(REDIRECT.getStatus()));
+            return new StringRepresentation(REDIRECT.getBody());
+          });
+    }
+  }
+
+  public static class CaptureHeadersResource extends BaseResource {
```
Comment (heyams) [inline instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java:103]:
```suggestion
  static class CaptureHeadersResource extends BaseResource {
```

File: instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+                    getResponse()
+                        .getAttributes()
+                        .computeIfAbsent("org.restlet.http.headers", (key) -> new Form());
+            responseHeaders.add("X-Test-Response", requestHeaders.getValues("X-Test-Request"));
+            return new StringRepresentation(CAPTURE_HEADERS.getBody());
+          });
+    }
+  }
+
+  public static class IndexedChildResource extends BaseResource {
```
Comment (heyams) [inline instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java:123]:
```suggestion
  static class IndexedChildResource extends BaseResource {
```

**Review by heyams — APPROVED** [review]

File: instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java (hunk 2 above)
Comment (jaydeluca) [reply instrumentation/restlet/restlet-1.1/testing/src/main/java/io/opentelemetry/instrumentation/restlet/v1_1/RestletAppTestBase.java:73]: these need to be public so jetty can access the constructors. When testing them as package private i see the following:


> 18:26:18.461 [qtp1342212468-54] WARN  org.eclipse.jetty.util.log - Exception while instantiating the target handler.
java.lang.NoSuchMethodException: io.opentelemetry.instrumentation.restlet.v1_1.RestletAppTestBase$QueryParamResource.<init>()


and for `PathParamResource`


> 18:27:52.806 [qtp111210112-54] WARN  org.eclipse.jetty.util.log - Exception while instantiating the target handler.
java.lang.NoSuchMethodException: io.opentelemetry.instrumentation.restlet.v1_1.RestletAppTestBase$PathParamResource.<init>()

etc

**Review by jaydeluca — APPROVED** [review]

**Pull request merged**